Description: Check that _Float16 is available before using it in openblas_config.h
 Fixes development package on armhf, s390x and sparc64.
Origin: upstream, https://github.com/OpenMathLib/OpenBLAS/commit/dc32a8a90f6e9169a6444826fcd55e0645db708c
                  https://github.com/OpenMathLib/OpenBLAS/commit/e926bb0523f1d001948a2214f213ec766d5d4414
Bug: https://github.com/OpenMathLib/OpenBLAS/issues/5747
Reviewed-by: Sébastien Villemot <sebastien@debian.org>
Last-Update: 2026-04-08
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/openblas_config_template.h
+++ b/openblas_config_template.h
@@ -43,7 +43,14 @@ typedef uint16_t bfloat16;
 #if defined(OPENBLAS_ARCH_POWER) || defined(OPENBLAS_ARCH_LOONGARCH64)
 typedef bfloat16 hfloat16;
 #else
+#define __STDC_WANT_IEC_60559_TYPES_EXT__
+#include <float.h>
+#ifdef FLT16_MAX
 typedef _Float16 hfloat16;
+#else
+#include <stdint.h>
+typedef uint16_t hfloat16;
+#endif
 #endif
 #else
 #include <stdint.h>
